dashboard-frontend/src/pages/StudyPage/Table.tsx

```diff
@@ -0,0 +1,190 @@
+import { classNames } from '../../helpers/strings';
+
+interface PaginationProps {
+  from: number;
+  to: number;
+  total: number;
+  previousButton: {
+    disabled: boolean;
+    onClick: () => void;
+  };
+  nextButton: {
+    disabled: boolean;
+    onClick: () => void;
+  };
+}
+
+const Table = ({
+  columnNames,
+  rows,
+  pagination,
+}: {
+  columnNames: string[];
+  rows: {
+    id: string;
+    firstColumn: string;
+    secondColumn: string;
+    thirdColumn: string;
+    fourthColumn: string;
+  }[];
+  pagination: PaginationProps;
+}) => (
+  <TableContainer pagination={pagination}>
+    <thead className="bg-gray-50">
+      <tr>
+        {columnNames.map(columnName => (
+          <TableHeaderCell key={columnName}>{columnName}</TableHeaderCell>
+        ))}
+      </tr>
+    </thead>
+    <tbody className="bg-white divide-y divide-gray-200">
+      {rows.map(row => (
+        <tr key={row.id}>
+          <TableCell first>{row.firstColumn}</TableCell>
+          <TableCell>{row.secondColumn}</TableCell>
+          <TableCell>{row.thirdColumn}</TableCell>
+          <TableCell>{row.fourthColumn}</TableCell>
+        </tr>
+      ))}
+    </tbody>
+  </TableContainer>
+);
+
+export const TableSkeleton = ({
+  numRows,
+  testId,
+  numColumns = 4,
+}: {
+  numRows: number;
+  testId?: string;
+  numColumns?: number;
+}) => (
+  <TableContainer
+    testId={testId}
+    pagination={{
+      from: 1,
+      to: numRows,
+      total: numRows,
+      previousButton: {
+        disabled: true,
+        onClick: () => {},
+      },
+      nextButton: {
+        disabled: true,
+        onClick: () => {},
+      },
+    }}
+  >
+    <thead className="bg-gray-50">
+      <tr>
+        {Array.from({ length: numColumns }, (_, index) => (
+          <TableHeaderCell key={index}>&nbsp;</TableHeaderCell>
+        ))}
+      </tr>
+    </thead>
+    <tbody className="bg-white divide-y divide-gray-200">
+      {Array.from({ length: numRows }, (_, index) => (
+        <tr key={index}>
+          <TableCell first>&nbsp;</TableCell>
+          <TableCell>&nbsp;</TableCell>
+          <TableCell>&nbsp;</TableCell>
+          <TableCell>&nbsp;</TableCell>
+        </tr>
+      ))}
+    </tbody>
+  </TableContainer>
+);
+
+const TableContainer = ({
+  testId,
+  children,
+  pagination,
+}: {
+  testId?: string;
+  children: React.ReactNode;
+  pagination: PaginationProps;
+}) => (
+  <div className="flex flex-col" data-testid={testId}>
+    <div className="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8">
+      <div className="py-2 align-middle inline-block min-w-full sm:px-6 lg:px-8">
+        <div className="shadow overflow-hidden border-b border-gray-200 rounded-lg">
+          <table className="min-w-full divide-y divide-gray-200">
+            {children}
+          </table>
+          <Pagination {...pagination} />
+        </div>
+      </div>
+    </div>
+  </div>
+);
+
+const Pagination = ({
```

Comment:
I'm suprised you had to build a table component from scratch. Tailwind doesn't give you one?

In any case, they look and work well, so I'm happy with them! And I like being able to customize their styling more in the future because they're custom. 

That being said, we'd definitely like to add the ability to sort by column at some point. Maybe also change the number viewed per page. It's all pretty traditional stuff, and shouldn't be hard to implement, but worth keeping in mind.

Reply:
I'm glad that you liked them 🙌

Answering your question, TailwindUI provides me with the raw separated JSX for the table and the pagination, the rest is up to me.

I wanted to have simpler version ready before going for the sorting, but we can definitely add it up soon. 👍

Reply:
Cool cool, thanks!